src/models/event.ts

```diff
@@ -107,7 +107,7 @@ export interface IEventRelation {
     event_id?: string;
     is_falling_back?: boolean;
     "m.in_reply_to"?: {
-        event_id: string;
+        event_id?: string;
```

Comment:
I'm also confused about this change, as the spec implies it's required

Reply:
It's required for a valid relation but someone can choose to include it or not.

And judging by the other fields above being optional, this one also seems optional.

Reply:
I believe the other fields such as `rel_type` are only optional because of replies being a special case. However in the case of replies, we know for certain that the `event_id` subkey is required, so I don't really see the point of letting invalid relations typecheck

Reply:
There is no guarantee the the field will be present. AFAIK, we don't have any validation to make sure this is true. The only place we use this type is for the following:

https://github.com/matrix-org/matrix-js-sdk/blob/df2b65f111cf09df19b12fbe8210a54a1a23b3d8/src/models/event.ts#L1298-L1303

There is no guarantee that `m.in_reply_to` -> `event_id` is present in downstream code and our downstream code should treat it that way.

Reply:
I see, arguably that method needs to do more validation, but I guess this is fine for now